R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,384 @@
+      mutate(value = value.x * CONV_DAYS_YEAR * (1 / CONV_MCAL_PCAL ) / value.y) %>%
+      select(-value.x, -value.y) ->
+      L134.pcFood_kcald_R_Dmnd_Y
+
+    # In preparation for future extrapolation, get the last historical numbers and aggregate by ISO
+    # Original lines 58-69
+    L100.FAO_ag_Food_t %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "crops") ->
+      L134.FAO_ag_Food_t
+
+    L100.FAO_an_Food_t %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      mutate(GCAM_demand = "meat") %>%
+      bind_rows(L134.FAO_ag_Food_t) %>%
+      left_join_keep_first_only(select(AGLU_ctry, FAO2050_reg, iso), by = "iso") ->
+      L134.Food_t_ctry_Dmnd_Y
+
+    # Calculate the future demand ratios by FAO2050 region for each demand type
+    # Drop unnecessary composite regions from FAO diet table
+    # Original lines 71-85
+    FAO2050_Diet %>%
+      filter(FAO2050_reg %in% L134.Food_t_ctry_Dmnd_Y$FAO2050_reg) %>%
+      # Calculate FAO2050 diet for specified diet years
+      # Use linear interpolation to convert FAO2050 model time periods to GCAM "diet years"
+      group_by(FAO2050_reg, FAO2050_item) %>%
+      # Extend years in dataset to include DIET_YEARS
+      complete(year = unique(c(.$year, aglu.DIET_YEARS))) %>%
+      mutate(value = approx_fun(year, value)) %>%
+      # Add caloric content and demand category
+      left_join(select(FAO2050_items_cal, FAO2050_item, GCAM_demand, kcalkg, conv_d), by = "FAO2050_item") ->
+      L134.Diet_Rfao_Cfao_Yfao
+
+    # Build new table with only diet years subsetted
+    # Multiply through by caloric contents to get all measures in kcal/pers/d, by FAO region and food categories
+    # Original lines 87-92
+    L134.Diet_Rfao_Cfao_Yfao %>%
+      filter(year %in% aglu.DIET_YEARS) %>%
+      mutate(value = value * kcalkg / conv_d) %>%
+      select(FAO2050_reg, FAO2050_item, GCAM_demand, year, value) ->
+      L134.Diet_kcald_Rfao_Cfao_Y
+
+    # Aggregate by GCAM demand and FAO region, and compute future diet ratios by FAO2050 region
+    # Original lines 94-104
+    L134.Diet_kcald_Rfao_Cfao_Y %>%
+      filter(year %in% aglu.DIET_YEARS, !is.na(GCAM_demand)) %>%
+      group_by(FAO2050_reg, GCAM_demand, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      mutate(GCAM_demand = if_else(GCAM_demand == "total", "crops", GCAM_demand)) ->
+      L134.Diet_kcald_Rfao_Dmnd_Y
+
+    # Note on this next section from Page:
+    # Can't just use exogenous caloric contents for the crop and meat commodities, which we're estimating
+    # as averages without knowing the exact composition of the commodities (e.g., what portion of "cereals"
+    # is corn vs wheat in each region, as they have different caloric contents), and still hit the target
```

Comment:
What we are trying to do is to get the FAO's projections of future dietary trends for two groupings: crops and meat.
We can't know exactly where their split is in the base year because this FAO AT2050 table just uses a few food groupings that are far coarser than the FAOSTAT data underlying these reported trends, and (generally) reports the flows in kilograms per person per day. While the total is reported in kca/pers/d, ideally we'd this information for all of groupings, or at least for crop and meat subtotals.